cmake/IWYU.cmake

```diff
@@ -0,0 +1,36 @@
+find_program(IWYU_PATH iwyu_tool)
+
+if(IWYU_PATH)
+    # iwyu is available
+
+    # add a target to inspect number of times headers are included.
+    add_custom_target(iwyu-list-includes
+        COMMAND grep -hor "\"#include .*\""  ${PROJECT_SOURCE_DIR}/src | sort | uniq -c | sort -n
+    )
+
+    # find the standard library directory
+    file(GLOB_RECURSE LLVM_INCLUDE /usr/lib*/llvm*/**/stddef.h)
```

Comment:
I'd not use GLOB_RECURSE here. Finding a directory should be possible via `find_path`.

Reply:
I tried and failed. While `find_path` has some support for wildcards, it is very limited and does not cover this case. Even if we turn `lib*` into `lib` (and `lib64`), I still could not get it to work.

Reply:
The IWYU documentation suggests to use `clang -print-resource-dir` to find the correct clang directory (https://github.com/include-what-you-use/include-what-you-use#how-to-install).